packages/http/src/mocker/negotiator/responseFinder.ts

```diff
@@ -0,0 +1,52 @@
+import { IHttpOperationResponse } from '@stoplight/types';
+import { head } from 'fp-ts/lib/Array';
+import { flatten, fold, none, Option, some } from 'fp-ts/lib/Option';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { tail } from 'lodash';
+import { Logger } from 'pino';
+import { createResponseFromDefault, findResponseByStatusCode } from './InternalHelpers';
+
+const msg = (code: number) => `Unable to find a ${code} response definition`;
+const defaultMsg = (code: number) => `Created a ${code} from a default response`;
+
+export const matchingOrder = [401, 403, 422, 400];
+
+export default function matchResponse(
+  httpResponses: IHttpOperationResponse[],
+  logger: Logger,
+  order: number[] = matchingOrder,
+): Option<IHttpOperationResponse> {
+  return pipe(
+    head(order),
+    fold(
+      () => flatten(tryToFindDefaultResponse(httpResponses, logger)),
+      possibleMatch => {
+        return pipe(
+          findResponseByStatusCode(httpResponses, `${possibleMatch}`),
+          fold(
```

Comment:
You do not need to use `fold` if you're not changing the `some` part — you should instead use a `alt` to provide an alternative option.

```ts
const t = pipe(
   trySomething,
   fold(()=> alternative, val => some(val))
)
```

-> becomes

```ts
const t = pipe(
   trySomething,
   alt(()=> alternative)
)
```

In this specific case

```ts
        return pipe(
          findResponseByStatusCode(httpResponses, `${possibleMatch}`),
          alt(() => {
              logger.trace(msg(possibleMatch));
              return matchResponse(httpResponses, logger, tail(order));
          }),
        );
```